flash/data/data_module.py

```diff
@@ -53,20 +56,35 @@ class DataModule(pl.LightningDataModule):
 
     def __init__(
         self,
-        train_dataset: Optional[Dataset] = None,
```

Comment:
It should still work if users provide their own dataset.

Reply:
Do we want to make this DataModule Flash specific ?

Reply:
I vote yes, otherwise can be confusing between lightning/flash components and their namings

Reply:
Have semi-reverted this so datamodule can now be made from datasets again